### PR TITLE
Feature/aql subquery execution block impl execute implementation

### PR DESCRIPTION
**Pull request by mchacki**
First implementation to pass through the AqlCall stacks.

This branch has an issue with profiles, as we do overfetch now, if we do precise fetching, other profiler tests fail.
We need to discuss hardLimit propagation as this would avoid the overmatching issues.

### CODE REVIEW

**Comment by mchacki** [comment]
Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7591/

File: tests/Aql/DependencyProxyMock.cpp [hunk 1]
```diff
@@ -21,6 +21,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 #include "DependencyProxyMock.h"
+#include <Logger/LogMacros.h>
```
Comment (hkernbach) [inline tests/Aql/DependencyProxyMock.cpp]: Remove me. 
```suggestion

```

File: tests/Aql/FilterExecutorTest.cpp [hunk 2]
```diff
@@ -387,23 +387,23 @@ TEST_F(FilterExecutorTest, test_produce_datarange_need_more) {
   SharedAqlItemBlockPtr inBlock =
       buildBlock<1>(itemBlockManager,
                     {{R"(true)"}, {R"(false)"}, {R"(true)"}, {R"(false)"}, {R"(true)"}});
-
+  size_t hardLimit = 1000;
```
Comment (hkernbach) [inline tests/Aql/FilterExecutorTest.cpp]: Maybe keep that 1000 in a more "organized" way? Right now it is not super obvious.

Comment (hkernbach) [reply tests/Aql/FilterExecutorTest.cpp]: Ah, ok. It's just a test, nevermind.

File: tests/Aql/SortLimit-test.cpp [hunk 3]
```diff
@@ -113,7 +112,7 @@ class SortLimitTest
     return strategy;
   }
 
-  bool verifyExpectedResults(TRI_vocbase_t& vocbase, std::string const& queryString,
+  void verifyExpectedResults(TRI_vocbase_t& vocbase, std::string const& queryString,
```
Comment (hkernbach) [inline tests/Aql/SortLimit-test.cpp]: :+1:

**Comment by mchacki** [comment]
Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9014/

**Comment by mchacki** [comment]
This branch has been created by multiple subbranches, which have been reviewed individually.
No further review required.

**Pull request merged**